### PR TITLE
ovl_En_Goroiwa OK

### PR DESCRIPTION
**Pull request by Sirius902**
This one was a lot but I got it. :)
I had to use `func_80042244` in one of my functions (@mzxrules has it decompiled in #256) so for now I added the prototype to `functions.h`.

### CODE REVIEW

File: include/functions.h
Comment (Dragorn421) [inline include/functions.h:600]: I don't think we use `bool`

File: src/overlays/actors/ovl_Bg_Spot01_Objects2/z_bg_spot01_objects2.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Spot01_Objects2/z_bg_spot01_objects2.c:115]: not a big issue but probably avoid modifying things not related to the PR

File: src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c:23]: prototypes for functions which are defined before being used such as `func_80A4C814` can be omitted

File: src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c:74]: `.list[0].` makes more sense than `.list->` imo

File: src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c:448]: does `0x10000` or `(f32)0x10000` work instead of `65536.0f`

File: src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c:487]: can `i * 0x4E20` be used instead of angle1?

File: src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c:672]:
```suggestion
```

File: src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c:728]: did you try a switch for this?

File: undefined_syms.txt
Comment (Dragorn421) [inline undefined_syms.txt:800]: `z_bg_jya_goroiwa` already defines another symbol used by `z_en_goroiwa`: `D_060006B0 = 0x060006B0;`

I don't know what the policy on this is, should the symbol be defined twice if that's possible?

**Review by Dragorn421 — CHANGES_REQUESTED** [review]
Good job 👍 
This actor is a nice demonstration of path usage

Is the rock height/diameter hardcoded anywhere? I guess in the collider but I also expected it to do something like `actorPos.y = path.y + height / 2`, can't locate such a thing

File: src/overlays/actors/ovl_Bg_Spot01_Objects2/z_bg_spot01_objects2.c
Comment (Sirius902) [reply src/overlays/actors/ovl_Bg_Spot01_Objects2/z_bg_spot01_objects2.c:115]: Fig said it was alright.

File: src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c
Comment (Sirius902) [reply src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c:483]: I don't believe so. I've tried that and couldn't get it to match.

File: src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c
Comment (Sirius902) [reply src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c:444]: Both work. Would `0x10000` be preferred?

File: undefined_syms.txt
Comment (Sirius902) [reply undefined_syms.txt:800]: The way it was explained to me was that you only added a symbol if it wasn't elsewhere but somebody correct me if I'm wrong.

File: src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c
Comment (Dragorn421) [reply src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c:444]: I think so, because it's normalizing the s16 angle `this->actor.shape.rot.y` which we use hex for

File: src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c:569]:
```suggestion
        if (yawDiff > -0x4000 && yawDiff < 0x4000) {
```

File: src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c:549]: can a `this = THIS` temp be afforded at all?

File: src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c:694]:
```suggestion
```

**Review by Dragorn421 — CHANGES_REQUESTED** [review]
very small things

File: src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c
Comment (Sirius902) [reply src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c:549]: Nope.

File: src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c:505]: `gravity` is more likely `-860` or something close if I got the math off,
and `minVelocityY` `-15000`

File: src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c:597]: merge this into a `else if` ?

**Review by Dragorn421 — CHANGES_REQUESTED** [review]
more small things

**Review by Dragorn421 — APPROVED** [review]

File: undefined_syms.txt
Comment (Roman971) [reply undefined_syms.txt:800]: Since both actors use the same object, D_060006B0 is the same symbol so there is no need to define it twice.

File: src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c
Comment (Roman971) [inline src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c:518]: Is `pathIdx` required to match here?

Comment (Sirius902) [reply src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c:518]: So I just messed with it a bit and I can get it to match if I use `(u8)this->actor.params` in place of `pathIdx`.

**Review by Roman971 — APPROVED** [review]
Looks good to me, tho I haven't looked at names at all so I can't say anything about that part.

File: src/overlays/actors/ovl_Bg_Spot01_Objects2/z_bg_spot01_objects2.c
Comment (fig02) [reply src/overlays/actors/ovl_Bg_Spot01_Objects2/z_bg_spot01_objects2.c:115]: yea just fyi @Dragorn421 stuff like this is more than fine. some people forget to run the formatter all the time, so when it runs on the whole codebase it might catch things. and its good to include 
(sorry for the 20 day late response)

File: src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c
Comment (fig02) [inline src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c:245]: gonna still merge, but just as a general note we dont seperate out static and non static variables in the list. sometimes they dont always fit at the top of the list and it doesnt make sense to treat them separately. if you do end up fixing this before i get to a merge, its relevant in a few other places in the file :D

**Review by fig02 — APPROVED** [review]

**Pull request merged**
